app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -87,13 +92,19 @@ def update_puppet_path
         path
       end
 
-      def pulp_node
-        @pulp_node ||= Katello::Pulp::Server.config(pulp_url, User.remote_user)
+      def update_pulp_url
+        if has_feature?(PULP_FEATURE)
```

Comment:
This needs an `else` and perhaps some way of dealing with the situation where the `url` is not found. For example when a proxy hasn't been updated.